raft/raft_test.go

```diff
@@ -288,33 +288,80 @@ func TestProgressPaused(t *testing.T) {
 }
 
 func TestLeaderElection(t *testing.T) {
```

Comment:
I think the cleanest way to handle this is with go 1.7 subtests. Do you still support go 1.6 or can you require go 1.7 features? I can change this to either use subtests or a separate helper function depending on what you require.